### PR TITLE
Add coarse dust product for RRFS

### LINKED ISSUE #638: Introduce coarse dust output for RRFS
A coarse dust variable has been added for RRFS, and this field needs to be output in UPP.  In addition, this coarse dust should impact the AOD.

### LINKED ISSUE #642: Update RRFS control file with removing some legacy fields
The RRFS_A and RRFS_B parallel runs have been using UPP control file postxconfig-NT-fv3lam_rrfs.txt from UPP/parm.  It indicates that the following fields are written in GRIB2 files from UPP with all missing values due to no output from model files in netcdf:
```
dataset: NATLEV:
AKHS:surface:2-3 hour ave fcst
AKMS:surface:2-3 hour ave fcst
PRES:surfacee:2-3 hour ave fcst
TMP:10 m above ground:2-3 hour ave fcst
TMP:10 m above ground:3 hour fcst
PRES:1-21 hybrid layer
PRES:1-61 hybrid layer

dataset: PRSLEV
RIME:1 hybrid level
LRGHR:entire atmosphere (considered as a single layer):0-3 hour ave fcst:
```
If no downstream applications will use them, can we remove from the UPP control file for RRFS?

### LINKED ISSUE #653: Restrict undefined grids in UPP computation
It is found that the following code might have undefined grids involved in computation:
https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/INITPOST_NETCDF.f#L2863
https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/INITPOST_NETCDF.f#L2875

@EricJames-NOAA Do you think the changes of restricting undefined grids in computation should be added?

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
This PR enables output of the coarse dust species from RRFS (coarsepm), analogous to previously added smoke and dust.  In addition, smoke, fine dust, and coarse PM now have aerosol size specifications (<2.5 microns for smoke and fine dust, and between 2.5 and 10 microns for coarse PM).  

One other modification made was to directly read aodtot from FV3 for RRFS, instead of calculating it in CALPW.f as was done for RAP/HRRR.

The code was tested on Jet for the RRFS_CONUS_3km system.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Could you sync your branch with the latest UPP develop? I will process your PR.

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Could you sync your branch with the latest UPP develop? I will process your PR.

@WenMeng-NOAA OK I synced it!  Thanks.

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA It appears to me that UPP IDs 800-889 are used for output simulated satellite products (See [here](https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f#L1749)) though no variable definitions in post_avblflds.xml. Could you reassign your new dust variables(801-804) starting from 1011? We might remove those legacy satellite products from UPP later. Thanks!

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA It appears to me that UPP IDs 800-889 are used for output simulated satellite products (See [here](https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f#L1749)) though no variable definitions in post_avblflds.xml. Could you reassign your new dust variables(801-804) starting from 1011? We might remove those legacy satellite products from UPP later. Thanks!

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Do you have the sample model history files in netcdf which include two new variables aodtot and coarsepm for my testing?

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Do you have the sample model history files in netcdf which include two new variables aodtot and coarsepm for my testing?

@WenMeng-NOAA I have copied some test files here on Hera: /scratch1/BMC/wrfruc/ejames/wen_coarsepm/

**Comment by WenMeng-NOAA** [comment]
> > @EricJames-NOAA Do you have the sample model history files in netcdf which include two new variables aodtot and coarsepm for my testing?
> 
> @WenMeng-NOAA I have copied some test files here on Hera: /scratch1/BMC/wrfruc/ejames/wen_coarsepm/

@EricJames-NOAA My test results are at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2023022318-new on Hera. I would think two new variables (ID 1014, 735) added in the control file postxconfig-NT-fv3lam_rrfs.txt are generated in PRSLEV dataset as:
```
828:717314305:d=2023022318:COLMD:entire atmosphere (considered as a single layer):3 hour fcst:aerosol=Dust Dry:aerosol_size >=2.5e-06,<1e-05:
760:674082656:d=2023022318:MASSDEN:8 m above ground:3 hour fcst:aerosol=Dust Dry:aerosol_size >=2.5e-06,<1e-05:
```
Please let me know if you see anything inconsistent with your test.

**Comment by EricJames-NOAA** [comment]
> > > @EricJames-NOAA Do you have the sample model history files in netcdf which include two new variables aodtot and coarsepm for my testing?
> > 
> > 
> > @WenMeng-NOAA I have copied some test files here on Hera: /scratch1/BMC/wrfruc/ejames/wen_coarsepm/
> 
> @EricJames-NOAA My test results are at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2023022318-new on Hera. I would think two new variables (ID 1014, 735) added in the control file postxconfig-NT-fv3lam_rrfs.txt are generated in PRSLEV dataset as:
> 
> ```
> 828:717314305:d=2023022318:COLMD:entire atmosphere (considered as a single layer):3 hour fcst:aerosol=Dust Dry:aerosol_size >=2.5e-06,<1e-05:
> 760:674082656:d=2023022318:MASSDEN:8 m above ground:3 hour fcst:aerosol=Dust Dry:aerosol_size >=2.5e-06,<1e-05:
> ```
> 
> Please let me know if you see anything inconsistent with your test.

@WenMeng-NOAA the tests look good to me!  The new fields look identical to what I obtained.  Thank you!

File: tests/compile_upp.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ usage() {
 }
 
 prefix="../install"
-ifi_opt=" -DBUILD_WITH_IFI=OFF"
+ifi_opt=" -DBUILD_WITH_IFI=ON"
```
Comment (WenMeng-NOAA) [inline tests/compile_upp.sh]: @EricJames-NOAA I prefer to set building without IFI as default for general UPP users since IFI code/lib is restricted.

Comment (EricJames-NOAA) [reply tests/compile_upp.sh]: Sorry about that, I did not mean to commit that change (it was just for my testing).  I have reverted it now.

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on WCOSS2, Hera and Orion with no changed results. This PR is ready for merging.

**Review by WenMeng-NOAA — APPROVED** [review]

**Pull request merged**